lib/ruby2d/window.rb

```diff
@@ -236,6 +223,36 @@ def add_object(o)
         false
       end
     end
-    
+
+    def events_store(event_type)
```

Comment:
I prepared a small benchmark:

```
require "benchmark/ips"

class CaseVsMetaprogramming
  def initialize
    @key = 0
    @key_down = 1
    @key_held = 2
    @key_up = 3
    @mouse = 4
    @mouse_up = 5
    @mouse_down = 6
    @mouse_scroll = 7
    @mouse_move = 8
    @controller = 9
    @controller_axis = 10
    @controller_button_down = 11
    @controller_button_up = 12
  end

  def metaprogramming(method)
    instance_variable_get "@#{method}"
  end

  def case_method(method)
    case method
    when :key
      @key
    when :key_down
      @key_down
    when :key_held
      @key_held
    when :key_up
      @key_up
    when :mouse
      @mouse
    when :mouse_up
      @mouse_up
    when :mouse_down
      @mouse_down
    when :mouse_scroll
      @mouse_scroll
    when :mouse_move
      @mouse_move
    when :controller
      @controller
    when :controller_axis
      @controller_axis
    when :controller_button_down
      @controller_button_down
    when :controller_button_up
      @controller_button_up
    end
  end
end

instance = CaseVsMetaprogramming.new

Benchmark.ips do |x|
  x.report("case") do
    instance.case_method(:key)
    instance.case_method(:key_down)
    instance.case_method(:key_held)
    instance.case_method(:key_up)
    instance.case_method(:mouse)
    instance.case_method(:mouse_up)
    instance.case_method(:mouse_down)
    instance.case_method(:mouse_scroll)
    instance.case_method(:mouse_move)
    instance.case_method(:controller)
    instance.case_method(:controller_axis)
    instance.case_method(:controller_button_down)
    instance.case_method(:controller_button_up)
  end

  x.report("meta") do
    instance.metaprogramming(:key)
    instance.metaprogramming(:key_down)
    instance.metaprogramming(:key_held)
    instance.metaprogramming(:key_up)
    instance.metaprogramming(:mouse)
    instance.metaprogramming(:mouse_up)
    instance.metaprogramming(:mouse_down)
    instance.metaprogramming(:mouse_scroll)
    instance.metaprogramming(:mouse_move)
    instance.metaprogramming(:controller)
    instance.metaprogramming(:controller_axis)
    instance.metaprogramming(:controller_button_down)
    instance.metaprogramming(:controller_button_up)
  end

  x.compare!
end
```

And the results are:

```
Warming up --------------------------------------
                case   124.819k i/100ms
                meta    22.528k i/100ms
Calculating -------------------------------------
                case      1.639M (± 3.7%) i/s -      8.238M in   5.033754s
                meta    247.047k (± 1.3%) i/s -      1.239M in   5.016220s

Comparison:
                case:  1639073.5 i/s
                meta:   247046.9 i/s - 6.63x  slower
```

But it's worth mentioning this is not a hot path. I would go for it, as long as we write somehere in docs that people shouldn't bind and unbind events in loop.